objects/src/transaction/tx_witness.rs

```diff
@@ -0,0 +1,147 @@
+    }
+
+    /// Returns a vector of [ConsumedNoteInfo] for all of the consumed notes in the transaction.
+    ///
+    /// # Errors
+    /// - If the consumed notes data is not found in the advice map.
+    /// - If the consumed notes data is not well formed.
+    pub fn consumed_notes_info(&self) -> Result<Vec<ConsumedNoteInfo>, TransactionWitnessError> {
+        // fetch consumed notes data from the advice map
+        let notes_data = self
+            .advice_witness
+            .mapped_values(&self.consumed_notes_hash.as_bytes())
+            .ok_or(TransactionWitnessError::ConsumedNoteDataNotFound)?;
+
+        // extract the notes from the first fetch and instantiate a vector to hold
+        // [ConsumedNoteInfo].
+        let num_notes = notes_data[0].as_int();
+        let mut consumed_notes_info = Vec::with_capacity(num_notes as usize);
+
+        // iterate over the notes and extract the nullifier and script root
+        let mut note_ptr = 1;
+        while note_ptr < notes_data.len() {
+            // make sure there is enough data to read (note data is well formed)
+            if note_ptr + 5 * WORD_SIZE > notes_data.len() {
+                return Err(TransactionWitnessError::InvalidConsumedNoteDataLength);
+            }
+
+            // compute the nullifier and script root
+            let nullifier =
+                Hasher::hash_elements(&notes_data[note_ptr..note_ptr + (4 * WORD_SIZE)]);
+            let script_root: Word = notes_data[note_ptr + WORD_SIZE..note_ptr + (2 * WORD_SIZE)]
+                .try_into()
+                .expect("word is well formed");
+
+            // push the [ConsumedNoteInfo] to the vector
+            consumed_notes_info.push(ConsumedNoteInfo::new(nullifier, script_root.into()));
+
+            // advance the pointers to the next note
+            let num_assets = notes_data[note_ptr + (4 * WORD_SIZE)].as_int();
+            note_ptr += (((num_assets + 1) & !1) as usize * WORD_SIZE) + 30;
+        }
+
+        debug_assert_eq!(
+            self.consumed_notes_hash,
+            Hasher::hash_elements(
+                &consumed_notes_info
+                    .iter()
+                    .flat_map(|info| <[Felt; 8]>::from(*info))
+                    .collect::<Vec<_>>()
+            )
+        );
+
+        Ok(consumed_notes_info)
+    }
+
+    /// Returns the transaction script root.
+    pub fn tx_script_root(&self) -> Option<Digest> {
+        self.tx_script_root
+    }
+
+    /// Returns the transaction [Program].
+    pub fn program(&self) -> &Program {
+        &self.program
+    }
+
+    /// Returns the stack inputs for the transaction.
+    pub fn stack_inputs(&self) -> StackInputs {
+        let mut inputs: Vec<Felt> = Vec::with_capacity(13);
+        inputs.extend(*self.consumed_notes_hash);
+        inputs.extend(*self.initial_account_hash);
+        inputs.push(*self.account_id);
+        inputs.extend(*self.block_hash);
+        StackInputs::new(inputs)
+    }
```

Comment:
minor nit: I'd rename this to `get_stack_inputs()` to indicate that we are actually building stack inputs in this method. (I usually use accessors without `get_` prefix when we are just returning a reference to an internal property).